rpcserver.go

```diff
@@ -5494,3 +5588,77 @@ func (r *rpcServer) BakeMacaroon(ctx context.Context,
 
 	return resp, nil
 }
+
+// FundingStateStep is an advanced funding related call that allows the caller
+// to either execute some preparatory steps for a funding workflow, or manually
+// progress a funding workflow. The primary way a funding flow is identified is
+// via its pending channel ID. As an example, this method can be used to
+// specify that we're expecting a funding flow for a particular pending channel
+// used to interactively drive PSBT signing for funding for partially complete
+// funding transactions.
+func (r *rpcServer) FundingStateStep(ctx context.Context,
+	in *lnrpc.FundingTransitionMsg) (*lnrpc.FundingStateStepResp, error) {
+
+	switch {
+
+	// If this is a message to register a new shim that is an external
+	// channel point, then we'll contact the wallet to register this new
+	// shim. A user will use this method to register a new channel funding
+	// workflow which has already been partially negotiated outside of the
+	// core protocol.
+	case in.GetShimRegister() != nil &&
+		in.GetShimRegister().GetChanPointShim() != nil:
+
+		rpcShimIntent := in.GetShimRegister().GetChanPointShim()
+
+		// Using the rpc shim as a template, we'll construct a new
+		// chanfunding.Assembler that is able to express proper
+		// formulation of this expected channel.
+		shimAssembler, err := newFundingShimAssembler(
```

Comment:
Don't we need the same basic sanity switch statement here too? Or better move that check into `newFundingShimAssembler`.

Reply:
Good point! Moved the sanity checks into `newFundingShimAssembler`.